app/src/main/java/edu/nextstep/camp/calculator/MainActivity.kt

```diff
@@ -1,22 +1,47 @@
 package edu.nextstep.camp.calculator
 
 import android.os.Bundle
+import android.view.View
 import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
 import edu.nextstep.camp.calculator.databinding.ActivityMainBinding
+import edu.nextstep.camp.calculator.view.MemoryDiffCallback
+import edu.nextstep.camp.calculator.view.MemoryRecyclerViewAdapter
+import edu.nextstep.camp.calculator.view.MemoryUIModel
 
 class MainActivity : AppCompatActivity(), MainContract.View {
     private lateinit var binding: ActivityMainBinding
 
     private lateinit var presenter: MainContract.Presenter
 
+    private lateinit var memoryAdapter: MemoryRecyclerViewAdapter
+
+    private val diffUtil = MemoryDiffCallback()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
         presenter = MainPresenter(this)
 
+        initView()
+    }
+
+    private fun initView() {
+        initRecyclerView()
+        setClickListener()
+    }
+
+    private fun initRecyclerView() {
+        binding.recyclerView.apply {
+            adapter = MemoryRecyclerViewAdapter()
+            memoryAdapter = adapter as MemoryRecyclerViewAdapter
+        }
```

Comment:
apply 스코프에는 수신 객체의 프로퍼티만 접근하는 것이 바람직합니다.
memoryAdatper가 수신객체의 프로퍼티라고 착각할 수 있기 때문이에요 :)

캐스팅을 또 한 번 하지 않고, 아래 처럼 작성해 볼 수도 있겠네요 :)
```suggestion
        memoryAdapter = MemoryRecyclerViewAdapter()
            .also { binding.recyclerView.adapter = it }
```

Reply:
오 감사합니다 하나 배워갑니다